src/agent/solana/exporter.rs

```diff
@@ -321,41 +429,15 @@ impl Exporter {
                     {
                         true // max delay since last published state reached, we publish anyway
                     } else {
-                        !last_info.cmp_no_timestamp(*info) // Filter out if data is unchanged
+                        !last_info.cmp_no_timestamp(info) // Filter out if data is unchanged
                     }
                 } else {
                     true // No prior data found, letting the price through
                 }
             })
             .collect::<Vec<_>>();
 
-        let publish_keypair = if let Some(kp) = self.key_store.publish_keypair.as_ref() {
-            // It's impossible to sanely return a &Keypair in the
-            // other if branch, so we clone the reference.
-            Keypair::from_bytes(&kp.to_bytes())
-                .context("INTERNAL: Could not convert keypair to bytes and back")?
-        } else {
-            // Request the keypair from remote keypair loader.  Doing
-            // this here guarantees that the up to date loaded keypair
-            // is being used.
-            //
-            // Currently, we're guaranteed not to clog memory or block
-            // the keypair loader under the following assumptions:
-            // - The Exporter publishing loop waits for a publish
-            //   attempt to finish before beginning the next
-            //   one. Currently realized in run()
-            // - The Remote Key Loader does not read channels for
-            //   keypairs it does not have. Currently expressed in
-            //   handle_key_requests() in remote_keypair_loader.rs
-
-            debug!(
-                self.logger,
-                "Exporter: Publish keypair is None, requesting remote loaded key"
-            );
-            let kp = RemoteKeypairLoader::request_keypair(&self.keypair_request_tx).await?;
-            debug!(self.logger, "Exporter: Keypair received");
-            kp
-        };
```

Comment:
Why did this get deleted

Reply:
This is not deleted. I moved it to another function so I can reuse it.